sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -323,9 +327,20 @@ object ParquetReadSupport {
    * @return A list of clipped [[GroupType]] fields, which can be empty.
    */
   private def clipParquetGroupFields(
-      parquetRecord: GroupType, structType: StructType, caseSensitive: Boolean): Seq[Type] = {
+      parquetRecord: GroupType,
+      structType: StructType,
+      caseSensitive: Boolean,
+      columnIndexAccess: Boolean): Seq[Type] = {
     val toParquet = new SparkToParquetSchemaConverter(writeLegacyParquetFormat = false)
-    if (caseSensitive) {
+    if (columnIndexAccess) {
+      structType.zipWithIndex.map { case (f, i) =>
+        if (i >= parquetRecord.getFieldCount) {
+          toParquet.convertField(f)
```

Comment:
shouldn't we call `clipParquetType`?

Reply:
addressed, but shall be applied in `else` branch